packages/remix/src/plugins/plugin.ts

```diff
@@ -0,0 +1,212 @@
+import { projectGraphCacheDirectory } from 'nx/src/utils/cache-directory';
+import {
+  type CreateDependencies,
+  type CreateNodes,
+  type CreateNodesContext,
+  type TargetConfiguration,
+  detectPackageManager,
+  readJsonFile,
+  writeJsonFile,
+} from '@nx/devkit';
+import { calculateHashForCreateNodes } from '@nx/devkit/src/utils/calculate-hash-for-create-nodes';
+import { getNamedInputs } from '@nx/devkit/src/utils/get-named-inputs';
+import { getLockFileName, getRootTsConfigPath } from '@nx/js';
+import { registerTsProject } from '@nx/js/src/internal';
+import { type AppConfig } from '@remix-run/dev';
+import { join, dirname } from 'path';
+import { existsSync, readdirSync } from 'fs';
+
+const cachePath = join(projectGraphCacheDirectory, 'remix.hash');
+const targetsCache = existsSync(cachePath) ? readTargetsCache() : {};
+const calculatedTargets: Record<
+  string,
+  Record<string, TargetConfiguration>
+> = {};
+
+function readTargetsCache(): Record<
+  string,
+  Record<string, TargetConfiguration>
+> {
+  return readJsonFile(cachePath);
+}
+
+function writeTargetsToCache(
+  targets: Record<string, Record<string, TargetConfiguration>>
+) {
+  writeJsonFile(cachePath, targets);
+}
+
+export const createDependencies: CreateDependencies = () => {
+  writeTargetsToCache(calculatedTargets);
+  return [];
+};
+
+export interface RemixPluginOptions {
+  buildTargetName?: string;
+  serveTargetName?: string;
+  startTargetName?: string;
+  typecheckTargetName?: string;
+}
+
+export const createNodes: CreateNodes<RemixPluginOptions> = [
+  '**/remix.config.{js,cjs}',
```

Comment:
Does remix support `.mjs` as well?

Reply:
I have not seen it in any of their examples or in the wild as of yet. 

They force `type: module` in the package.json which allows .js file extensions to be used.

I will test out if .mjs is supported by the current remix compiler, and if so I can update this glob

Reply:
.mjs is supported